app/upgrades/v11/upgrades.go

```diff
@@ -74,3 +94,86 @@ func MigrateEscrowAccounts(ctx sdk.Context, ak authkeeper.AccountKeeper) {
 		ak.SetModuleAccount(ctx, acc)
 	}
 }
+
+// HandleRewardDistribution handles the logic for the reward distribution,
+// it only commits to the db if successful
+func HandleRewardDistribution(ctx sdk.Context, bk bankkeeper.Keeper,
+	sk stakingkeeper.Keeper, dk distributionkeeper.Keeper, logger log.Logger,
+) {
+	// use a cache context as a rollback mechanism in case
+	// the distrbution fails
+	cacheCtx, writeFn := ctx.CacheContext()
+	err := DistributeRewards(cacheCtx, bk, sk, dk)
+	if err != nil {
+		// log error instead of aborting the upgrade
+		logger.Error("failed to distribute rewards", "error", err.Error())
+	} else {
+		writeFn()
+	}
+}
+
+// DistributeRewards distributes the token allocations from the Olympus Mons
+// incentivized testnet for completing the Mars Meteor Missions
+func DistributeRewards(ctx sdk.Context, bk bankkeeper.Keeper, sk stakingkeeper.Keeper,
+	dk distributionkeeper.Keeper,
+) error {
+	funder := sdk.MustAccAddressFromBech32(FundingAccount)
+	numValidators := sdk.NewInt(int64(len(Validators)))
+
+	for _, allocation := range Allocations {
+		// send reward to receiver
+		receiver := sdk.MustAccAddressFromBech32(allocation[0])
+		receivingAmount, ok := sdk.NewIntFromString(allocation[1])
+		if !ok {
+			return errorsmod.Wrapf(errortypes.ErrInvalidType,
+				"cannot retrieve allocation from string for address %s",
+				allocation[0],
+			)
+		}
+		reward := sdk.Coins{sdk.NewCoin(types.BaseDenom, receivingAmount)}
+		err := bk.SendCoins(ctx, funder, receiver, reward)
+		if err != nil {
+			return err
+		}
+
+		// delegate receiver's rewards to selected validators equally
+		delegationAmt := reward.QuoInt(numValidators)[0].Amount
+		remainderAmount := (reward[0].Amount).Mod(numValidators)
+		for i, validatorBech32 := range Validators {
+			validatorAddress, err := sdk.ValAddressFromBech32(validatorBech32)
+			if err != nil {
+				return err
+			}
+			validator, found := sk.GetValidator(ctx, validatorAddress)
+			if !found {
+				return errorsmod.Wrapf(slashingtypes.ErrBadValidatorAddr,
+					"validator address %s cannot be found",
+					validatorAddress)
+			}
+			// we delegate the remainder to the first validator, for the sake of testing
+			// consistency. this remainder is in the order of 10^-18 evmos, and at most
+			// 10^-15 evmos after all rewards are allocated
+			if remainderAmount.IsPositive() && i == 0 {
```

Comment:
I'd just transfer it to the community pool

Reply:
same as above, removed this logic entirely